### PR TITLE
Strip input from both space and non-breaking space

### PR DESCRIPTION
**Pull request by dentarg**
Related to #126, making the whitespace removal more general.

WDYT @walro @jage?

### CODE REVIEW

**Comment by walro** [comment]
Haven't checked the logics, but I like the gist of it.

File: lib/twingly/url.rb [hunk 1]
```diff
@@ -68,14 +75,14 @@ def internal_parse(input)
...
       end
 
-      def strip_nbsp(input)
+      def strip_whitespace(input)
         return input unless input.encoding == Encoding::UTF_8
 
-        input = input.sub(/\A#{NBSP}/, "")
-        input = input.sub(/#{NBSP}\z/, "")
+        input = input.sub(/\A[#{WHITESPACE}]+/, "")
+        input = input.sub(/[#{WHITESPACE}]+\z/, "")
```
[the remark this answers is not in the document]
Comment (dentarg) [reply lib/twingly/url.rb]: Does the change from `\A` to `^` matter in your diff?

File: lib/twingly/url.rb [hunk 2]
```diff
@@ -23,13 +23,20 @@ class URL
       PublicSuffix::DomainInvalid,
     ].freeze
     NBSP = "\u00A0"
+    SPACE = "\u0020"
+    WHITESPACE = [
+      NBSP,
+      SPACE,
+    ].join
```
[the remark this answers is not in the document]
Comment (dentarg) [reply lib/twingly/url.rb]: > Only string literals get automatic freeze

Ah, okay.

[the remark this answers is not in the document]
Comment (dentarg) [reply lib/twingly/url.rb]: Yeah, why not

File: lib/twingly/url.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (walro) [reply lib/twingly/url.rb]: Oops, my mistake. I tried some things and forgot to change it back completely. I believe `\A` should still work.

[the remark this answers is not in the document]
Comment (jage) [reply lib/twingly/url.rb]: Just a note, if we go this route I think the regexp should be in a constant, to enable reuse of it.

**Comment by jage** [comment]
Looks like a good direction!

File: lib/twingly/url.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (dentarg) [reply lib/twingly/url.rb]: 31a56d3

[the remark this answers is not in the document]
Comment (dentarg) [reply lib/twingly/url.rb]: 77d54ad

File: lib/twingly/url.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (dentarg) [reply lib/twingly/url.rb]: 77d54ad

File: lib/twingly/url.rb [hunk 3]
```diff
@@ -23,13 +23,20 @@ class URL
       PublicSuffix::DomainInvalid,
     ].freeze
     NBSP = "\u00A0"
+    SPACE = "\u0020"
+    WHITESPACE = [
```
[the remark this answers is not in the document]
Comment (dentarg) [reply lib/twingly/url.rb]: 77d54ad

File: lib/twingly/url.rb [hunk 4]
```diff
@@ -23,13 +23,22 @@ class URL
       PublicSuffix::DomainInvalid,
     ].freeze
     NBSP = "\u00A0"
+    SPACE = "\u0020"
+    WHITESPACE_CHARS = [
+      NBSP,
+      SPACE,
+    ].join.freeze
+    WHITESPACE = /\A[#{WHITESPACE_CHARS}]+|[#{WHITESPACE_CHARS}]+\z/.freeze
```
Comment (walro) [inline lib/twingly/url.rb]: `LEADING_AND_TRAILING_WHITESPACE`... short and concise :D

Comment (dentarg) [reply lib/twingly/url.rb]: ae9e161

**Pull request merged**